tests/test_vonkarman.py

```diff
@@ -0,0 +1,209 @@
+
+
+@timer
+def test_vk_ne():
+    gsp = galsim.GSParams(maxk_threshold=1.1e-3, folding_threshold=5.1e-3)
+
+    objs = [galsim.VonKarman(lam=500.0, r0=0.2, L0=25.0),
+            galsim.VonKarman(lam=500.0, r0=0.2, L0=25.0, flux=2.2),
+            galsim.VonKarman(lam=500.0, r0=0.2, L0=1e11),
+            galsim.VonKarman(lam=500.0, r0=0.1, L0=20.0),
+            galsim.VonKarman(lam=550.0, r0=0.1, L0=20.0),
+            galsim.VonKarman(lam=550.0, r0=0.1, L0=20.0, do_delta=True),
+            galsim.VonKarman(lam=550.0, r0=0.1, L0=20.0, scale_unit=galsim.arcmin),
+            galsim.VonKarman(lam=550.0, r0=0.1, L0=20.0, gsparams=gsp)]
+    all_obj_diff(objs)
+
+
+@timer
+def test_vk_eq_kolm():
+    lam = 500.0
+    r0 = 0.2
+    L0 = 3e5  # Need to make this surprisingly large to make vk -> kolm.
+    flux = 3.3
+    kolm = galsim.Kolmogorov(lam=lam, r0=r0, flux=flux)
+    vk = galsim.VonKarman(lam=lam, r0=r0, L0=L0, flux=flux)
+
+    np.testing.assert_allclose(kolm.xValue(0,0), vk.xValue(0,0), rtol=1e-3, atol=0)
+
+    kolm_img = kolm.drawImage(nx=24, ny=24, scale=0.2)
+    vk_img = vk.drawImage(nx=24, ny=24, scale=0.2)
+    np.testing.assert_allclose(kolm_img.array, vk_img.array, atol=flux*1e-5, rtol=0)
+
+
+@timer
+def test_vk_fitting_formulae():
+    #         lam, r0_500, L0
+    params = [(650, 0.15, 10.0),
+              (450, 0.12, 25.0),
+              (900, 0.18, 100.0)]
+
+    def predicted_FWHM_ratio(r0, L0):
+        """Fitting formula for VonKarman FWHM / Kolmogorov FWHM
+        from Martinez++2014
+        """
+        return np.sqrt(1 - 2.183*(r0/L0)**0.356)
+
+    def predicted_HLR_ratio(r0, L0):
+        """Fitting formula for VonKarman HLR / Kolmogorov HLR
+        from Martinez++2014
+        """
+        return np.sqrt(1 - 1.534*(r0/L0)**0.347)
+
+    for lam, r0_500, L0 in params:
+        print(lam, r0_500, L0)
+        r0 = r0_500*(lam/500.0)**(6./5)
+        kolm = galsim.Kolmogorov(lam=lam, r0=r0)
+        vk = galsim.VonKarman(lam=lam, r0=r0, L0=L0)
+        HLR_ratio = vk.calculateHLR() / kolm.calculateHLR()
+        FWHM_ratio = vk.calculateFWHM() / kolm.calculateFWHM()
+        print(HLR_ratio)
+        print(FWHM_ratio)
+        np.testing.assert_allclose(HLR_ratio, predicted_HLR_ratio(r0, L0), rtol=0.015)
+        np.testing.assert_allclose(FWHM_ratio, predicted_FWHM_ratio(r0, L0), rtol=0.015)
+
+
+def vk_benchmark():
+    import time
+    t0 = time.time()
+    vk = galsim.VonKarman(lam=700, r0=0.1, L0=24.3)
+    vk.drawImage(nx=16, ny=16, scale=0.2)
+    t1 = time.time()
+    print("Time to create/draw first time: {:6.3f}s".format(t1-t0))
+    for i in range(10):
+        vk.drawImage(nx=16, ny=16, scale=0.2)
+    t2 = time.time()
+    print("Time to draw 10 more: {:6.3f}s".format(t2-t1))
+    for i in range(100):
+        vk.drawImage(nx=16, ny=16, scale=0.2, method='phot', n_photons=50000)
+    t3 = time.time()
+    print("Time to photon-shoot 100 more with 50000 photons each: {:6.3f}s".format(t3-t2))
```

Comment:
Worth putting a summary of what you discovered with this benchmark test somewhere in here?

Reply:
Sure.  I added comments inline with the print statements showing the speeds on my laptop:
~0.7s for the first vk, ~0.07s for the next 10, and ~0.9s for drawing 100x50,000 photons.